cumulus_library/studies/core/core_templates/allergyintolerance.sql.jinja

```diff
@@ -0,0 +1,138 @@
+{% import 'core_utils.jinja' as utils %}
+{% import 'unnest_utils.jinja' as unnest_utils %}
+
+CREATE TABLE core__allergyintolerance AS
+WITH temp_allergyintolerance AS (
+    SELECT
+         {{-
+            utils.basic_cols(
+                'allergyintolerance',
+                'a',
+                [
+                    'id',
+                ]
+            )
+        }},
+        {{-
+            utils.nullable_cols(
+                'allergyintolerance',
+                'a',
+                [
+                    'type',
+                    'category',
+                    'criticality',
+                    ('patient', 'reference', 'patient_ref'),
+                    ('encounter', 'reference', 'encounter_ref'),
+                ],
+                schema
+            )
+        }},
+        {{-
+            utils.date_cols_from_str(
+                'allergyintolerance',
+                'a',
+                ['recordedDate'],
+                schema
+            )
+        }},
+        {{-
+            utils.truncate_date_cols(
+                'allergyintolerance',
+                'a',
+                [
+                    ('recordedDate', 'week'),
+                    ('recordedDate', 'month'),
+                    ('recordedDate', 'year'),
+                ],
+                schema
+            )
+        }}
+    FROM allergyintolerance AS a
+),
+
+temp_category AS (
+    SELECT
+        a.id,
+        t.category
+    FROM
+        allergyintolerance AS a,
+        UNNEST(a.category) AS t (category)
+),
+
+flattened_reaction AS ({{ unnest_utils.flatten('allergyintolerance', 'reaction') }}),
+
+temp_reaction AS (
+    SELECT
+        r.id,
+        r.row,
+        dn_subs.code AS substance_code,
+        dn_subs.system AS substance_system,
+        dn_subs.display AS substance_display,
+        dn_man.code AS manifestation_code,
+        dn_man.system AS manifestation_system,
+        dn_man.display AS manifestation_display,
+        {{-
+            utils.nullable_cols(
+                'allergyintolerance',
+                'r',
+                [
+                    ('reaction', 'severity', 'severity'),
+                ],
+                schema
+            )
+        }}
+    FROM flattened_reaction AS r
+    LEFT JOIN core__allergyintolerance_dn_reaction_substance AS dn_subs
+        ON r.id = dn_subs.id AND r.row = dn_subs.row
+    LEFT JOIN core__allergyintolerance_dn_reaction_manifestation AS dn_man
+        ON r.id = dn_man.id AND r.row = dn_man.row
+)
+
+SELECT
```

Comment:
Open to feedback on which fields I've included! The [US Core spec](http://hl7.org/fhir/us/core/STU4/StructureDefinition-us-core-allergyintolerance.html) is wildly bare - not even an encounter field or a recordedDate timestamp.

Stuff like `category` is super useful! (I'd think? Maybe not - maybe all you care about is "am I gonna get sick from this thing")

Reply:
I like it until we're told otherwise.